operator/pkg/lbipam/lbipam.go

```diff
@@ -0,0 +1,1728 @@
+					}
+
+					poolSync = true
+					return nil
+				},
+				func(k resource.Key, clbi *cilium_api_v2alpha1.CiliumLoadBalancerIPPool) error {
+					err := ipam.handleNewPool(clbi)
+					if err != nil {
+						syncErrors = append(syncErrors, err)
+					}
+
+					return nil
+				},
+				func(k resource.Key, clbi *cilium_api_v2alpha1.CiliumLoadBalancerIPPool) error {
+					syncErrors = append(syncErrors, errors.New("Got a Pool deleted event while syncing"))
+
+					return nil
+				},
+			)
+		}
+	}
+
+	// Then sync all services
+	svcSync := false
+	for !svcSync {
+		select {
+		case event := <-svcChan:
+			event.Handle(
+				func() error {
+					svcSync = true
+
+					err := ipam.satisfyServices()
+					if err != nil {
+						syncErrors = append(syncErrors, err)
+						return nil
+					}
+
+					err = ipam.updateAllPoolCounts()
+					if err != nil {
+						syncErrors = append(syncErrors, err)
+						return nil
+					}
+
+					return nil
+				},
+				func(k resource.Key, svc *core_v1.Service) error {
+					err := ipam.handleUpsertService(svc)
+					if err != nil {
+						syncErrors = append(syncErrors, err)
+					}
+
+					return nil
+				},
+				func(k resource.Key, svc *core_v1.Service) error {
+					syncErrors = append(syncErrors, errors.New("Got a Pool deleted event while syncing"))
+
+					return nil
+				},
+			)
+		}
+	}
+
+	if len(syncErrors) > 0 {
+		return multierr.Combine(syncErrors...)
+	}
+
+	return nil
+}
+
+func (ipam *LBIPAM) poolOnUpsert(k resource.Key, pool *cilium_api_v2alpha1.CiliumLoadBalancerIPPool) error {
+	// Deep copy so we get a version we are allowed to update
+	pool = pool.DeepCopy()
+
+	var err error
+	if _, exists := ipam.poolStore.GetByName(pool.GetName()); exists {
+		ipam.logger.Debugf("Updating pool '%s'", pool.GetName())
+		err = ipam.handlePoolModified(pool)
+		if err != nil {
+			ipam.logger.WithError(err).Error("Error in handlePoolModified()")
+		}
+	} else {
+		ipam.logger.Debugf("Adding pool '%s'", pool.GetName())
+		err = ipam.handleNewPool(pool)
+		if err != nil {
+			ipam.logger.WithError(err).Error("Error in handleNewPool()")
+		}
+	}
+	if err != nil {
+		if isRetryableError(err) {
+			// Return the error, this will trigger the service resource to retry this service at a later time in the hope
+			// that any transient error is resolved by then.
+			return err
+		}
+
+		return nil
+	}
+
+	err = ipam.settleConflicts()
+	if err != nil {
+		ipam.logger.WithError(err).Error("Error in settleConflicts()")
+
+		if isRetryableError(err) {
+			// Return the error, this will trigger the service resource to retry this service at a later time in the hope
+			// that any transient error is resolved by then.
+			return err
+		}
+
+		return nil
+	}
+
+	err = ipam.satisfyAndUpdateCounts()
+	if err != nil {
+		ipam.logger.WithError(err).Error("Error in satisfyAndUpdateCounts()")
+
+		if isRetryableError(err) {
+			// Return the error, this will trigger the service resource to retry this service at a later time in the hope
```

Comment:
Comment talks about the "service resource". Perhaps just shorten the comment here to "return the error to retry later"?